index.d.ts

```diff
@@ -0,0 +1,35 @@
+import {Options} from 'p-map';
+
+export type Mapper<ValueType, KeyType, MappedValueType> = (
+	element: ValueType,
+	index: KeyType
```

Comment:
The `index` is always `number`

Reply:
Sorry? It defeinitely is the `key` from the hashmap/Map passed to p-props:
```js
await pMap(Object.entries(input), ([key, value]) => mapper(value, key), options);
```

Reply:
You're right. I was just confused about the naming.